File: Plot/src/main/java/io/deephaven/plot/PlotStyle.java
/*
 * Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
 */

package io.deephaven.plot;

import java.util.Arrays;

/**
 * The style of a plot (e.g. line, bar, etc.).
 */
public enum PlotStyle {
    /**
     * A bar chart.
     */
    BAR,

    /**
     * A stacked bar chart.
     */
    STACKED_BAR,

    /**
     * A line chart.
     *
     * Does not display shapes at data points by default.
     */
    LINE,

    /**
     * An area chart.
     *
     * Does not display shapes at data points by default.
     */
    AREA,

    /**
     * A stacked area chart.
     *
     * Does not display shapes at data points by default.
     */
    STACKED_AREA,

    /**
     * A pie chart.
     */
    PIE,

    /**
     * A histogram chart.
     */
    HISTOGRAM,

    /**
     * An open-high-low-close chart.
     */
    OHLC,

    /**
     * A scatter plot.
     *
     * Lines are not displayed by default.
     */
    SCATTER,

    STEP,

    /**
     * An error bar plot.
     *
     * Points are not displayed by default.
     */
    ERROR_BAR;


    /**
     * Returns the requested plot style.
     *
     * @param style case insensitive style name
     * @throws IllegalArgumentException {@code style} must not be null
     * @return PlotStyle with the given name
     */
    @SuppressWarnings("ConstantConditions")
    public static PlotStyle plotStyle(final String style) {
        if (style == null) {
            throw new IllegalArgumentException("PlotStyle can not be null!");
        }

        try {
            return PlotStyle.valueOf(style.toUpperCase());
        } catch (IllegalArgumentException e) {
            throw new IllegalArgumentException("PlotStyle " + style + " is not defined");
        }
    }

    /**
     * Returns the names of available plot styles.
     *
     * @return array of the PlotStyle names
     */
    public static String[] plotStyleNames() {
        return Arrays.stream(PlotStyle.values()).map(Enum::name).toArray(String[]::new);
    }

}

File: py/server/deephaven/plot/plotstyle.py
#
# Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
#

""" This module provides support for different plot styles (e.g. line, bar, area, etc.). """

from enum import Enum

import jpy

_JPlotStyle = jpy.get_type("io.deephaven.plot.PlotStyle")


class PlotStyle(Enum):
    """ An enum defining the styles of a plot (e.g. line, bar, etc.). """

    BAR = _JPlotStyle.BAR
    """ A bar chart. """

    STACKED_BAR = _JPlotStyle.STACKED_BAR
    """ A stacked bar chart. """

    LINE = _JPlotStyle.LINE
    """ A line chart (does not display shapes at data points by default). """

    AREA = _JPlotStyle.AREA
    """ An area chart (does not display shapes at data points by default). """

    STACKED_AREA = _JPlotStyle.STACKED_AREA
    """ A stacked area chart (does not display shapes at data points by default). """

    PIE = _JPlotStyle.PIE
    """ A pie chart. """

    HISTOGRAM = _JPlotStyle.HISTOGRAM
    """ A histogram chart. """

    OHLC = _JPlotStyle.OHLC
    """ An open-high-low-close chart. """

    SCATTER = _JPlotStyle.SCATTER
    """ A scatter plot (lines are not displayed by default). """

    STEP = _JPlotStyle.STEP
    """ A step plot. """

    ERROR_BAR = _JPlotStyle.ERROR_BAR
    """ An error bar plot (points are not displayed by default). """
